src/ContactComponent.php

```diff
@@ -92,44 +92,63 @@ function wf_crm_contact_search($node, $element, $params, $contacts, $str = NULL)
     $limit = $str ? 12 : 500;
     $ret = [];
     $display_fields = array_values($element['#results_display']);
-    $search_field = 'display_name';
+    $fieldMappings = [
+      'current_employer' => ['employer_id', 'display_name'],
+      'email' => ['email', 'email'],
+      'phone' => ['phone', 'phone'],
+      'city' => ['address', 'city'],
+      'state_province' => ['address', 'state_province_id:label'],
+      'country' => ['address', 'country_id:label'],
+      'county' => ['address', 'county_id:label'],
+      'postal_code' => ['address', 'postal_code'],
+    ];
+    $joinedTables = [];
+    foreach ($fieldMappings as $field => $type) {
+      if ($key = array_search($field, $display_fields)) {
+        unset($display_fields[$key]);
+        [$table, $fieldName]= $type;
+        $display_fields[] = "{$table}.{$fieldName}";
+        if (empty($joinedTables[$table]) && in_array($table, ['email', 'phone', 'address'])) {
+          $joinedTables[$table] = TRUE;
+          $params['join'][] = [ucfirst($table) . " AS {$table}", 'LEFT', ["{$table}.is_primary", '=', 1]];
+        }
+      }
+    }
     $sort_field = 'sort_name';
     // Search and sort based on the selected display field
     if (!in_array('display_name', $display_fields)) {
-      $search_field = $sort_field = $display_fields[0];
+      $sort_field = $display_fields[0];
     }
     $params += [
-      'rowCount' => $limit,
-      'sort' => $sort_field,
-      'return' => $display_fields,
+      'limit' => $limit,
+      'orderBy' => [
+        $sort_field => 'ASC',
+      ],
+      'select' => $display_fields,
     ];
     if (!empty($params['relationship']['contact'])) {
       $c = $params['relationship']['contact'];
       $relations = NULL;
       if (!empty($contacts[$c]['id'])) {
         $relations = $this->wf_crm_find_relations($contacts[$c]['id'], wf_crm_aval($params['relationship'], 'types'));
-        $params['id'] = ['IN' => $relations];
+        $params['where'][] = ['id', 'IN', $relations];
       }
       if (!$relations) {
         return $ret;
       }
     }
     unset($params['relationship']);
     if ($str) {
-      $str = str_replace(' ', '%', \CRM_Utils_Type::escape($str, 'String'));
-      // The contact api takes a quirky format for display_name and sort_name
-      if (in_array($search_field, ['sort_name', 'display_name'])) {
-        $params[$search_field] = $str;
-      }
-      // Others use the standard convention
-      else {
-        $params[$search_field] = ['LIKE' => "%$str%"];
+      $searchFields = [];
+      foreach ($display_fields as $fld) {
+        $searchFields[] = [$fld, 'CONTAINS', $str];
       }
+      $params['where'][] = ['OR', $searchFields];
     }
-    $result = $this->utils->wf_civicrm_api('contact', 'get', $params);
+    $result = $this->utils->wf_civicrm_api4('Contact', 'get', $params);
```

Comment:
The recent test fails seems to be from here, since api4 checks perms by default, so the existing contact dropdown doesn't have Default Org in it since the anon user doesn't have permission.

And then lower down in wf_crm_contact_access where the other api4 replacement was made, the logic will need to change to account for the change in default. And as written currently it's assuming all callers to wf_crm_contact_access will have been updated to pass in filters['checkPermissions'] instead of filters['check_permissions']